_source/_posts/2023-07-31-workflows-workshop.md

```diff
@@ -0,0 +1,939 @@
+1. Click **Unnamed** (upper left corner)
+1. For the name, enter **Todo Report**
+1. It is also a good idea to enter a description **This flow sends automated Todo report email.** 
+1. Check the **Save all data that passes through the Flow?** checkbox. You will be using this capability later in this tutorial
+1. Click **Save** to save the flow name and description
+
+{% img blog/workflows-workshop/Workflows_create_new_flow.jpg alt:"Creating new flow" %}{: .center-image }
+
+Congratulations, you created a Workflows flow. 
+
+{% img blog/workflows-workshop/Workflows_new_flow_created.jpg alt:"A new Workflows flow" %}{: .center-image }
+
+## Workflows building blocks
+Before building the flow, you should learn about Workflows building blocks. 
+
+Each card represents one step in a flow. 
+
+A card has two types of fields: input and output. Input fields accept values. Output fields are values a card produces, which can be passed to subsequent cards.
+
+This is an example of the **Text - Length** card. This card calculates and returns the number of characters in a text string.
+
+- The **text** field is the card's input
+- The **length** field is the card's output (the lower grey area)
+
+{% img blog/workflows-workshop/Workflows_Text_Length_card.jpg alt:"The Text-Length card calculates text length" %}{: .center-image }
+
+There are two types of cards:
+
+- App action cards are steps that control other applications or web services. For example, Gmail, Slack, and Jira
+- Function cards are steps to interact with, change, or control the data in a flow. Some examples of functions cards:
+    - True/False: Evaluate values based on true or false conditions
+    -  Flow Control: Manage and manipulate the structure of your flow
+    -  Date & Time: Parse and manipulate times and dates
+    -  List: Create and iterate over lists of items
+    -  Number: Perform mathematical operations
+    -  Text: Build, modify, and parse text
+
+
+Here is again what you are going to build: 
+
+- Calling an API service to retrieve all the todo items
+- Creating a message from all the todo items
+- Emailing the message
+- Set up the flow to run on schedule
+
+Before building the flow, you need to get the Todo application.  
+
+## Setting up the Todo application
+In this section, you will complete the following steps:  
+
+1. Getting the Todo application
+2. Installing application dependencies
+3. Creating the database
+4. Adding an organziation to the application
+5. Running the application
+
+
+### Getting the Todo application
+In this step, you will get the Todo application.
+
+There are two ways to get the Todo application: 
+
+  _Option 1: Clone the repository_
+  1. Clone the [Todo application](https://github.com/oktadev/okta-enterprise-ready-workshops/) that already has OIDC support implemented. 
+  2. Run the following command `git checkout oidc-workshop-complete`
+  
+  _Option 2: Download the application_
```

Comment:
There should be some explanation that reader can download the completed application for this tutorial to use with their workflow or something

Reply:
@alisaduncan do you want me to add more information about option #2 (Download the application)?

Reply:
Like what Nick has in his post right after the Prereqs here
https://developer.okta.com/blog/2022/07/25/express-prisma-tailwind

That way people know there's code for the completed workflows project if they need to compare, or start directly from that step and only add the workflows, etc.